src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.cs

```diff
@@ -224,26 +225,41 @@ async ValueTask DecrementInFlightCountAsync(InFlightSolution inFlightSolution)
             AssetProvider assetProvider,
-            Checksum solutionChecksum,
+            Checksum newSolutionChecksum,
             CancellationToken cancellationToken)
         {
             try
             {
                 // Try to create the solution snapshot incrementally off of the workspaces CurrentSolution first.
                 var updater = new SolutionCreator(Services.HostServices, assetProvider, this.CurrentSolution);
-                if (await updater.IsIncrementalUpdateAsync(solutionChecksum, cancellationToken).ConfigureAwait(false))
+                if (await updater.IsIncrementalUpdateAsync(newSolutionChecksum, cancellationToken).ConfigureAwait(false))
                 {
-                    return await updater.CreateSolutionAsync(solutionChecksum, cancellationToken).ConfigureAwait(false);
+                    return await updater.CreateSolutionAsync(newSolutionChecksum, cancellationToken).ConfigureAwait(false);
                 }
                 else
                 {
                     // Otherwise, this is a different solution, or the first time we're creating this solution.  Bulk
                     // sync over all assets for it.
-                    await assetProvider.SynchronizeSolutionAssetsAsync(solutionChecksum, cancellationToken).ConfigureAwait(false);
+                    await assetProvider.SynchronizeSolutionAssetsAsync(newSolutionChecksum, cancellationToken).ConfigureAwait(false);
 
                     // get new solution info and options
-                    var solutionInfo = await assetProvider.CreateSolutionInfoAsync(solutionChecksum, cancellationToken).ConfigureAwait(false);
-                    return CreateSolutionFromInfo(solutionInfo);
+                    var solutionInfo = await assetProvider.CreateSolutionInfoAsync(newSolutionChecksum, cancellationToken).ConfigureAwait(false);
+                    var solution = CreateSolutionFromInfo(solutionInfo);
+
+                    // ensure that the solution has the correct source generator execution versions. note we should do
+                    // this all in a unified fashion with CreateSolutionAsync above.  However, that is blocked in
+                    // https://github.com/dotnet/roslyn/pull/72860
+                    {
+                        var newSolutionCompilationChecksums = await assetProvider.GetAssetAsync<SolutionCompilationStateChecksums>(
+                            AssetPathKind.SolutionCompilationStateChecksums, newSolutionChecksum, cancellationToken).ConfigureAwait(false);
+
+                        var newVersions = await assetProvider.GetAssetAsync<SourceGeneratorExecutionVersionMap>(
+                            AssetPathKind.SolutionSourceGeneratorExecutionVersionMap, newSolutionCompilationChecksums.SourceGeneratorExecutionVersionMap, cancellationToken).ConfigureAwait(false);
+
+                        solution = solution.WithSourceGeneratorExecutionVersions(newVersions, cancellationToken);
+                    }
```

Comment:
Note: this is something that was missing.  but is also a stopgap.  the right way to do things is to do teh initial bulk sync, then run the normal 'delta' sync algorithm to ensure that all checksums match.  That's what https://github.com/dotnet/roslyn/pull/72860, but it has a problem i'm working with Jason on.